From 4e64b32feccabb6aaf05dcd9b4788fbfeef63a3f Mon Sep 17 00:00:00 2001
From: Binbin Zhou <zhoubinbin@loongson.cn>
Date: Tue, 31 Dec 2024 15:56:52 +0800
Subject: [PATCH 176/301] AOSCOS: drm/ls2kbmc: Add id_table for
 ls2kbmc_platform_driver

Try to fix device binding for the driver.

Signed-off-by: Binbin Zhou <zhoubinbin@loongson.cn>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/tiny/ls2kbmc.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/tiny/ls2kbmc.c b/drivers/gpu/drm/tiny/ls2kbmc.c
index 4b440f20cb4d..55c2b6568642 100644
--- a/drivers/gpu/drm/tiny/ls2kbmc.c
+++ b/drivers/gpu/drm/tiny/ls2kbmc.c
@@ -904,10 +904,17 @@ static void ls2kbmc_remove(struct platform_device *pdev)
 	drm_dev_unplug(dev);
 }
 
+static struct platform_device_id ls2kbmc_id_table[] = {
+	{ "ls2kbmc-framebuffer", },
+	{ /* sentinel */ }
+};
+MODULE_DEVICE_TABLE(platform, ls2kbmc_id_table);
+
 static struct platform_driver ls2kbmc_platform_driver = {
 	.driver = {
 		.name = "ls2kbmc-framebuffer",
 	},
+	.id_table = ls2kbmc_id_table,
 	.probe = ls2kbmc_probe,
 	.remove = ls2kbmc_remove,
 };
-- 
2.48.1

